### PR TITLE
K8SPS-27: Add PMM sidecar to MySQL

### PR DESCRIPTION
**Pull request by egegunes**
TODO:
- [ ] Add tests
- [ ] Resources

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

commit: https://github.com/percona/percona-server-mysql-operator/pull/20/commits/980c3b1de3967a1aaa21571cd52dfdfc2c7a57fd
image: `perconalab/percona-server-mysql-operator:PR-20-980c3b1`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

commit: https://github.com/percona/percona-server-mysql-operator/pull/20/commits/2a3b6c99655a86530306c3a0097eed609a1fda22
image: `perconalab/percona-server-mysql-operator:PR-20-2a3b6c9`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

commit: https://github.com/percona/percona-server-mysql-operator/pull/20/commits/2a3b6c99655a86530306c3a0097eed609a1fda22
image: `perconalab/percona-server-mysql-operator:PR-20-2a3b6c9`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | failed |

commit: https://github.com/percona/percona-server-mysql-operator/pull/20/commits/99ae5d0e76b38b0b6c7a28467532f0437ea5ed04
image: `perconalab/percona-server-mysql-operator:PR-20-99ae5d0`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | failed |

commit: https://github.com/percona/percona-server-mysql-operator/pull/20/commits/6a686c141d18ba22b0f14b7c013558df262c8e97
image: `perconalab/percona-server-mysql-operator:PR-20-6a686c1`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |

commit: https://github.com/percona/percona-server-mysql-operator/pull/20/commits/d98d7e757eadfb1dcb1bb7c6fe4606affcef83e6
image: `perconalab/percona-server-mysql-operator:PR-20-d98d7e7`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | failed |

commit: https://github.com/percona/percona-server-mysql-operator/pull/20/commits/af7c89cb8c307f0b813db0715d8456af1005f114
image: `perconalab/percona-server-mysql-operator:PR-20-af7c89c`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |

commit: https://github.com/percona/percona-server-mysql-operator/pull/20/commits/a77f02f8bb067861b6e26c62e4ff1df3a4e65fb4
image: `perconalab/percona-server-mysql-operator:PR-20-a77f02f`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| monitoring | passed |

commit: https://github.com/percona/percona-server-mysql-operator/pull/20/commits/cd0ddce3ad34f4fd1fea2dc2bbc500d87c24ed41
image: `perconalab/percona-server-mysql-operator:PR-20-cd0ddce`

File: e2e-tests/init-deploy/00-assert.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+apiVersion: kuttl.dev/v1beta1
+kind: TestAssert
+timeout: 120
```
Comment (cap1984) [inline e2e-tests/init-deploy/00-assert.yaml:3]: is it necessary to wait up to 2 mins?

File: Jenkinsfile [hunk 2]
```diff
@@ -168,6 +169,18 @@ pipeline {
 
                     sudo sh -c "curl -s -L https://github.com/mikefarah/yq/releases/download/3.3.2/yq_linux_amd64 > /usr/local/bin/yq"
```
Comment (cap1984) [inline Jenkinsfile:170]: let's switch to yq 4

File: e2e-tests/init-deploy/00-deploy-operator.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+kind: TestStep
+commands:
+- script: |
+    . ../functions
+    set -o errexit
+    set -o xtrace
+
+    kubectl -n ${NAMESPACE} apply -f ${src_dir}/deploy/crd.yaml
+    kubectl -n ${NAMESPACE} apply -f ${src_dir}/deploy/rbac.yaml
+    yq --doc 1 w ${src_dir}/deploy/operator.yaml 'spec.template.spec.containers[0].image' "${IMAGE}" | kubectl -n ${NAMESPACE} apply -f -
```
Comment (cap1984) [inline e2e-tests/init-deploy/00-deploy-operator.yaml:11]: let's switch to yq 4

**Review by cap1984 — CHANGES_REQUESTED** [review]

File: deploy/secrets.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: cluster1-secrets
+type: Opaque
+data:
```
Comment (hors) [inline deploy/secrets.yaml:6]: please use `stringData` as in https://github.com/percona/percona-xtradb-cluster-operator/blob/main/deploy/secrets.yaml

File: e2e-tests/conf/secrets.yaml [hunk 5]
```diff
@@ -0,0 +1,25 @@
+apiVersion: v1
+kind: Secret
+type: Opaque
+metadata:
+  name: cluster1-secrets
+data:
```
Comment (hors) [inline e2e-tests/conf/secrets.yaml:6]: please use stringData as in https://github.com/percona/percona-xtradb-cluster-operator/blob/main/deploy/secrets.yaml

**Review by hors — CHANGES_REQUESTED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

commit: https://github.com/percona/percona-server-mysql-operator/pull/20/commits/980c3b1de3967a1aaa21571cd52dfdfc2c7a57fd
image: `perconalab/percona-server-mysql-operator:PR-20-980c3b1`

File: e2e-tests/init-deploy/00-assert.yaml (hunk 1 above)
Comment (egegunes) [reply e2e-tests/init-deploy/00-assert.yaml]: kuttl tries to assert and if there are assertion failures it sleeps a second and assert again until the timeout. we're saying fail the test if you can't assert this in 2 mins, so it's a worst case scenario.

I tuned the timeouts in https://github.com/percona/percona-server-mysql-operator/pull/21/commits/f7e76c345bd7838fa30efc2b5eb5215b0eeb96ee

File: Jenkinsfile (hunk 2 above)
Comment (egegunes) [reply Jenkinsfile]: https://github.com/percona/percona-server-mysql-operator/pull/21/commits/ce9dbd37615ce61ca476bb38d31829eadc785ddd

File: e2e-tests/conf/secrets.yaml (hunk 5 above)
Comment (egegunes) [reply e2e-tests/conf/secrets.yaml]: https://github.com/percona/percona-server-mysql-operator/pull/21/commits/8d6a0e91559771c3e2e21618eddc1c34fb559959

File: deploy/secrets.yaml (hunk 4 above)
Comment (egegunes) [reply deploy/secrets.yaml]: https://github.com/percona/percona-server-mysql-operator/pull/20/commits/b1ce183403382348f9cce75920e52b7ef593512c

File: e2e-tests/init-deploy/00-deploy-operator.yaml (hunk 3 above)
Comment (egegunes) [reply e2e-tests/init-deploy/00-deploy-operator.yaml]: https://github.com/percona/percona-server-mysql-operator/pull/21/commits/ce9dbd37615ce61ca476bb38d31829eadc785ddd

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

commit: https://github.com/percona/percona-server-mysql-operator/pull/20/commits/b1ce183403382348f9cce75920e52b7ef593512c
image: `perconalab/percona-server-mysql-operator:PR-20-b1ce183`

**Review by delgod — APPROVED** [review]

**Review by hors — APPROVED** [review]

**Review by cap1984 — APPROVED** [review]

**Pull request merged**